database/src/main/java/com/firebase/ui/database/FirebaseIndexArray.java

```diff
@@ -38,6 +38,9 @@
     private FirebaseArray<String> mKeySnapshots;
     private List<DataSnapshot> mDataSnapshots = new ArrayList<>();
 
+    private List<String> mQueuedAdditions = new ArrayList<>();
```

Comment:
Took me a little while to figure out what these variables do.  I'd suggest a rename and some comments.  How about this (and these are just my first stab so feel free to think up better ones):

```java
// Keys for which we have received an 'ADDED' event but have not yet downloaded the indexed data
private List<String> mKeysWithPendingData = new ArrayList();

// True if a key has been modified, moved, or deleted but we have not propagated the change yet. False otherwise.
private boolean mHasPendingModifyOrDelete;
```